Cargo.toml

```diff
@@ -34,7 +34,7 @@ serde = { version = "1.0", features = ["derive"], optional = true }
 serde_json = { version = "1.0", optional = true }
 
 [dev-dependencies]
-simple_logger = "2.1"
+simple_logger = "4.0"
```

Comment:
@alamb Would be reasonable to add a logging structure to the project?

By that I mean, logging each step of the parsing. That would make error tracing more feasible, without the need of changing the current error and API structure.

This could be handled by a logger, which level can be changed by caller crates (one example is the postgres rust crate [(1)](https://docs.rs/postgres/latest/postgres/index.html), which does that).

[1] : https://docs.rs/postgres/latest/postgres/index.html

Reply:
I think logging is very useful -- perhaps we could look into https://docs.rs/log/0.4.17/log/ which seems widely used across the rust ecosystem

Then `trace!` level logging could be used to for low level error tracing

Reply:
Ok! I'll start working on that as I work on other features :)